addons/edi/models/edi_document.py

```diff
@@ -310,13 +344,32 @@ def _get_state_name(self):
         vals = dict(self.fields_get(allfields=["state"])["state"]["selection"])
         return vals[self.state]
 
+    def _compute_edi_rec_count(self):
```

Comment:
I'm really not sure this should be computed.

This also smells a bit to me as it is something that has to be implemented on each EDI (as we can see by the amount of changes in these MRs), and it's another thing to remember to do each time we add a new EDI.

I think the intention with this was to basically do the same thing the logging is doing (i.e ` _logger.info("Executed %s in %.2fs, %d queries", self.name, stats.elapsed, stats.count)`) and then it only needs to be in one place. That is already handled by the way EDI records are batched in groups of 1,000's. That is the only level of granularity we require, as there's not much point in having lower granularity due to the speed in which rows are processed..

So, can't we just replace all of these `_compute_edi_rec_count` hooks to just one single hook just before/after that logger entry? There's also a similar one for preparing, in `public/odoo_edi/addons/edi/models/edi_document.py::action_execute() and ::action_prepare()`

Reply:
I think it's worth exploring if we can achieve more or less the intention of the story without having to implement a new computed method for each document type.